### ISSUE TITLE
ClayCSS: Update and add table utilities to allow for better sizing and alignment

### ISSUE THREAD

**Issue by pat270** [liferay/clay#1065]

**Comment by pat270** [comment]
Hey @marcoscv-work,

Portal's admin section has a lot of tables, some are 7 columns, 4 columns, 3 columns wide with varying content (a lot of user input we have no control over e.g., description, name, title, file name). This gives more control when used with `table-cell-expand` / `table-cell-expand-*`. There is also bug at https://issues.liferay.com/browse/LPS-83241.

**Comment by jbalsas** [comment]
Hey @pat270, can you also describe what the helpers do? I think that as we move forward improving the clay documentation, this should include some docs for the new helpers, so maybe these questions won't be needed :)

Seeing the code I think I get the idea, but please indulge me just for reference.

My initial doubt was that eventually, tables are going to be dynamic. We're going to allow users to show/hide and drag/drop columns, so the table structure won't be known in advance and might change dynamically. That is to say... helpers need to take into account this dynamic nature as I think they do in this case. Can you confirm how this should work in those scenarios?

**Comment by pat270** [comment]
These are column helpers, so the same classes will have to be applied to all td's in the same column. Dragging, dropping, hiding columns or rows wouldn't create a need to change class names. The only thing that would cause class name changes would be if there is support for dragging to make columns/rows bigger or smaller, which if that is the case, we wouldn't use any helper. Example markup below:

```
<table class="table table-autofit table-list>
  <thead>
    <tr>
      <th class="minw-150 table-cell-expand-small">Title</th>
      <th class="minw-400 table-cell-expand">Description</th>
      <th>Action</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td class="minw-150 table-cell-expand-small">My Book Title 1</td>
      <td class="minw-400 table-cell-expand">Some description</td>
      <td><button></button></td>
    </tr>
    <tr>
      <td class="minw-150 table-cell-expand-small">My Book Title 2</td>
      <td class="minw-400 table-cell-expand">Some description 2</td>
      <td><button></button></td>
    </tr>
  </tbody>
</table>
```

The action column will only be as wide as its content. The description column will fill remaining space but won't shrink below 400px. The title column will be 25% as wide as the description column but won't shrink below 150px.

**Comment by marcoscv-work** [comment]
Great! it will work perfectly! Thanks Patrick.

About the class names, I know bootstrap helpers is not a great example but they already have "`w-`limits" and "`mw-100`" classes in percent:
http://getbootstrap.com/docs/4.1/utilities/sizing/

They create the classes from an each from sizes. 
https://github.com/twbs/bootstrap/blob/v4-dev/scss/utilities/_sizing.scss

Anyway, this example is for general helper classes and we need them for tables, is it better if we use a `.table-*` modifier?

Thanks @pat270 !

**Comment by pat270** [comment]
@marcoscv-work I prefixed the table helpers. They are now `.table-cell-minw-*` and `.table-cell-ws-*`.

**Comment by matuzalemsteles** [comment]
hey guys, I'm closing this once it has already been fixed at #1066.
